environments/flow/optional.js

```diff
@@ -0,0 +1,32 @@
+/**
+ * Js-coding-standards
+ *
+ * @author      Robert Rossmann <rr.rossmann@me.com>
+ * @copyright   2016 STRV
+ * @license     http://choosealicense.com/licenses/bsd-3-clause  BSD-3-Clause License
+ */
+
+'use strict'
+
+module.exports = {
+  extends: './recommended.js',
+
+  rules: {
+    // Enforces a particular style for boolean type annotations.
+    'flowtype/boolean-style': ['warn', 'boolean'],
+
+    // Disallows use of primitive constructors as types, such as Boolean, Number and String.
+    'flowtype/no-primitive-constructor-types': 'warn',
+
+    // Warns against weak type annotations any, Object and Function. These types can cause flow
+    // to silently skip over portions of your code, which would have otherwise caused type errors.
+    'flowtype/no-weak-types': ['warn', { any: true }],
+
+    // Enforces sorting of Object annotations.
+    // This rule mirrors ESlint's sort-keys rule.
+    'flowtype/sort-keys': 'warn',
```

Comment:
Because after all changes `environments/flow/optional.js` contains only this rule and `flowtype/type-id-match` I would left them as optional part of ruleset.